src/plugins/intel_cpu/src/infer_request.cpp

```diff
@@ -558,36 +558,6 @@ void SyncInferRequest::init_tensor(const std::size_t& port_index, const ov::ISyn
                     tensor = ov::make_tensor(model_prec, tensor_shape);
                 }
                 ov::ISyncInferRequest::set_tensor(port, tensor);
-            } else {
-                const auto& tensor_shape = tensor->get_shape();
-                const bool isDynamic = port_shape.is_dynamic();
-                // Static shape case is enough information that shapes are incompatible to throw exception
-                // but in dynamic shape case we also need to handle following corner case:
-                // on tensor initialization stage we create empty tensor with dimensions equal 0
-                // so if we have tensor with all zero dimension we mustn't throw exception
-                if (!port_shape.compatible(ov::PartialShape(tensor_shape)) &&
-                    (!isDynamic || static_cast<int64_t>(tensor_shape.size()) != port_shape.rank().get_length() ||
-                     std::any_of(tensor_shape.begin(), tensor_shape.end(), [](const size_t& dims) {
-                         return dims != 0;
```

Comment:
Dynamic shape will create tensor with 0-dimention for dynamic dimension, for example:
   ` partial shape = [1..99,3,3]` will generate `[0,3,3]` tensor by: `OutputControlBlock control_block{model_prec, Shape{shape}};` 
After applying `id` to index port replacing `name`, it will go to this branch for the ports sharing tensor.
For dynamic shape tensor, i didn't see obvious issue when initialize it with 0-dimension.